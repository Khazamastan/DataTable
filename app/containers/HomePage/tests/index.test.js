/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import LanguageProvider from 'containers/LanguageProvider';
import { fetchSongs } from '../actions';
import { HomePage, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';
import { translationMessages } from '../../../i18n';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <LanguageProvider messages={translationMessages}>
          <IntlProvider locale="en" defaultLocale="en">
            <HomePage
              loading={false}
              error={false}
              query={{}}
              loadMoreRows={() => {}}
              totalCount={0}
              songs={[]}
            />
          </IntlProvider>
        </LanguageProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('fetchSongs', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.fetchSongsData).toBeDefined();
      });

      it('should dispatch fetchSongs when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.fetchSongsData();
        expect(dispatch).toHaveBeenCalledWith(fetchSongs());
      });
    });
  });
});
