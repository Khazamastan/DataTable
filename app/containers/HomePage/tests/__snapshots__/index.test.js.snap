// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<article>
  <div
    class="HomeWrapper-sc-1oq0eh2-0 kyJZWt"
  >
    <div>
      <section
        class="Section-vxhas0-0 pfTbc"
      >
        <div
          class="HomeWrapper__HeaderWrapper-sc-1oq0eh2-1 dOaGar"
        >
          <h2>
            <span>
              Data Table Component!
            </span>
          </h2>
        </div>
        <div
          class="table-container"
        >
          <div
            class="container"
          >
            <div
              class="SearchWrapper-q18sbo-0 fjVACF"
            >
              <div
                class="Wrapper-sc-1yw1li4-0 efyKhh"
              >
                <button
                  class="StyledButton-sc-1lrmbv8-0 lkmlui"
                >
                  Filter
                </button>
              </div>
              <input
                class="search Wrapper__InputWrapper-sc-15n0e2a-0 joPiQc"
                placeholder="Search here"
                type="text"
                value=""
              />
            </div>
            <p
              class="showing-text"
            >
              Showing 
              <strong>
                0
              </strong>
            </p>
            <div
              class="TableWrapper-sc-1eiqmgm-0 kWvpHK"
            >
              <div
                class="Wrapper-sc-12b7pcm-0 evwzjo"
              >
                <div>
                  <div
                    class=" checkbox Wrapper-uo5kd0-0 heHCKw"
                  >
                    <button
                      class="Wrapper__CheckBoxViewWrapper-uo5kd0-1 USkkw"
                    >
                      <input
                        type="checkbox"
                        value=""
                      />
                       
                      <span>
                        â–¾
                      </span>
                    </button>
                  </div>
                  <div
                    class="  Wrapper-uo5kd0-0 heHCKw"
                  >
                    <p
                      class="id"
                    >
                      ID
                       
                      
                    </p>
                  </div>
                  <div
                    class="  Wrapper-uo5kd0-0 fCILUl"
                  >
                    <p
                      class="thumbnailUrl"
                    >
                      
                       
                      
                    </p>
                  </div>
                  <div
                    class="  Wrapper-uo5kd0-0 jEHECx"
                  >
                    <p
                      class="sort"
                    >
                      Title
                       
                      
                    </p>
                  </div>
                  <div
                    class="numeric  Wrapper-uo5kd0-0 bGzQBi"
                  >
                    <p
                      class="sort"
                    >
                      Album ID
                       
                      
                    </p>
                  </div>
                  <div
                    class="  Wrapper-uo5kd0-0 bGzQBi"
                  >
                    <p
                      class="sort"
                    >
                      URL
                       
                      
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="Wrapper-sc-1766fc-0 joDhtK"
              >
                <div>
                  <div
                    style="overflow: visible; width: 0px;"
                  >
                    <div>
                      <div
                        aria-label="grid"
                        aria-readonly="true"
                        class="ReactVirtualized__Grid ReactVirtualized__List"
                        role="grid"
                        style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: hidden;"
                        tabindex="0"
                      />
                    </div>
                  </div>
                  <div
                    class="resize-triggers"
                  >
                    <div
                      class="expand-trigger"
                    >
                      <div
                        style="width: 1px; height: 1px;"
                      />
                    </div>
                    <div
                      class="contract-trigger"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</article>
`;
